### ISSUE TITLE
Auto upload: Existing files (conflicts) are not handled/resolved correctly

### ISSUE THREAD

**Issue by funkmaus** [nextcloud/android#6841]
### Steps to reproduce

1. Create a custom auto upload directive. Settings:
 - Upload existing files: Yes
 - What should be done if a file already exists?: Overwrite the version on the server.
2. Wait for the existing files to be uploaded successfully.
3. Change a file on the device and wait for the Nextcloud app to recognize that change.

Additional steps:

4. Resolve the resulting conflict by choosing "Which version do you want to keep? Local file".

### Expected behaviour

The Nextcloud app should upload the new version to the server (overwriting the old version on the server). Step 4 in the list above should not be necessary. This was the actual behaviour in versions prior to 3.13.0.

### Actual behaviour

 - After step 3: Nextcloud app gives me a notification about a synchronization conflict. This conflict is also listed under `Uploads -> Failed`. The new version is not uploaded to the server.

 - After step 4: Nextcloud app uploads the new version to the server (overwriting the old version on the server) but again gives me a notification about a synchronization conflict that is also listed under `Uploads -> Failed`.

### Can you reproduce this problem on https://try.nextcloud.com?

Yes it can be reproduced there.

### Environment data

Android version: 9, security patches from June 2020

Device model: Fairphone 3

Stock or customized system: Stock

Nextcloud app version: 3.13.0 (F-Droid)

Nextcloud server version: 19.0.1

Reverse proxy: None

#### Web server error log

No errors.

#### Web server access log

After step 3:
```
<IPAddr> - - [28/Aug/2020:09:31:33 +0200] "GET /<PathToNextcloud>/index.php/204 HTTP/1.1" 204 1671 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.13.0"
<IPAddr> - - [28/Aug/2020:09:31:33 +0200] "GET /<PathToNextcloud>/index.php/204 HTTP/1.1" 204 1675 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.13.0"
<IPAddr> - - [28/Aug/2020:09:31:33 +0200] "GET /<PathToNextcloud>/index.php/204 HTTP/1.1" 204 1675 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.13.0"
<IPAddr> - - [28/Aug/2020:09:31:33 +0200] "GET /<PathToNextcloud>/index.php/204 HTTP/1.1" 204 1675 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.13.0"
<IPAddr> - <User> [28/Aug/2020:09:31:33 +0200] "HEAD /<PathToNextcloud>/remote.php/webdav/<ServerAutoUploadDir>/<ChangedFile>.suffix HTTP/1.1" 200 1257 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.13.0"
<IPAddr> - - [28/Aug/2020:09:31:37 +0200] "GET /<PathToNextcloud>/index.php/204 HTTP/1.1" 204 1677 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.13.0"
<IPAddr> - <User> [28/Aug/2020:09:31:37 +0200] "HEAD /<PathToNextcloud>/remote.php/webdav/<ServerAutoUploadDir>/<ChangedFile>.suffix HTTP/1.1" 200 779 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.13.0"
```

After step 4:
```
<IPAddr> - - [28/Aug/2020:09:34:26 +0200] "GET /<PathToNextcloud>/index.php/204 HTTP/1.1" 204 4777 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.13.0"
<IPAddr> - <User> [28/Aug/2020:09:34:26 +0200] "HEAD /<PathToNextcloud>/remote.php/webdav/<ServerAutoUploadDir>/<ChangedFile>.suffix HTTP/1.1" 200 4363 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.13.0"
<IPAddr> - <User> [28/Aug/2020:09:34:27 +0200] "PUT /<PathToNextcloud>/remote.php/webdav/<ServerAutoUploadDir>/<ChangedFile>.suffix HTTP/1.1" 204 803 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.13.0"
<IPAddr> - <User> [28/Aug/2020:09:34:29 +0200] "PROPFIND /<PathToNextcloud>/remote.php/webdav/<ServerAutoUploadDir>/<ChangedFile>.suffix HTTP/1.1" 207 1489 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.13.0"
<IPAddr> - <User> [28/Aug/2020:09:34:29 +0200] "GET /<PathToNextcloud>/index.php/apps/files/api/v1/thumbnail/256/256/<ServerAutoUploadDir>/<ChangedFile>.suffix HTTP/1.1" 404 726 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.13.0"
```

#### Nextcloud log (data/nextcloud.log)
After step 3:
```
{"reqId":"nEt2QhZ9lmBDRJYHTxlH","level":3,"time":"2020-08-28T07:31:33+00:00","remoteAddr":"<IPAddr>","user":"--","app":"PHP","method":"GET","url":"/<PathToNextcloud>/index.php/204","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"TMcVcCfvu8KfPNonqOUT","level":3,"time":"2020-08-28T07:31:33+00:00","remoteAddr":"<IPAddr>","user":"--","app":"PHP","method":"GET","url":"/<PathToNextcloud>/index.php/204","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"d3T7ij2cKm8R0cATi3pB","level":3,"time":"2020-08-28T07:31:33+00:00","remoteAddr":"<IPAddr>","user":"--","app":"PHP","method":"GET","url":"/<PathToNextcloud>/index.php/204","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"d0o2WtW3jTyahkq54L1H","level":3,"time":"2020-08-28T07:31:33+00:00","remoteAddr":"<IPAddr>","user":"--","app":"PHP","method":"GET","url":"/<PathToNextcloud>/index.php/204","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"3Od3M5SuXzf5toeWFrsd","level":3,"time":"2020-08-28T07:31:33+00:00","remoteAddr":"<IPAddr>","user":"<User>","app":"PHP","method":"HEAD","url":"/<PathToNextcloud>/remote.php/webdav/<ServerAutoUploadDir>/<ChangedFile>.suffix","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"ooo2udAiheA2Fs45A9yT","level":3,"time":"2020-08-28T07:31:37+00:00","remoteAddr":"<IPAddr>","user":"--","app":"PHP","method":"GET","url":"/<PathToNextcloud>/index.php/204","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"1zY9k2ntunTDIix8iMYD","level":3,"time":"2020-08-28T07:31:37+00:00","remoteAddr":"<IPAddr>","user":"<User>","app":"PHP","method":"HEAD","url":"/<PathToNextcloud>/remote.php/webdav/<ServerAutoUploadDir>/<ChangedFile>.suffix","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
```

After step 4:
```
{"reqId":"g7xZrWiAzfzNJHopLBkr","level":3,"time":"2020-08-28T07:34:12+00:00","remoteAddr":"<IPAddr>","user":"<User>","app":"PHP","method":"PROPFIND","url":"/<PathToNextcloud>/remote.php/webdav/<ServerAutoUploadDir>/<ChangedFile>.suffix","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"9Jvhn7Rk9CXapq22ylhK","level":3,"time":"2020-08-28T07:34:26+00:00","remoteAddr":"<IPAddr>","user":"--","app":"PHP","method":"GET","url":"/<PathToNextcloud>/index.php/204","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"udcL6E1oumS8Wv2pChXF","level":3,"time":"2020-08-28T07:34:27+00:00","remoteAddr":"<IPAddr>","user":"<User>","app":"PHP","method":"HEAD","url":"/<PathToNextcloud>/remote.php/webdav/<ServerAutoUploadDir>/<ChangedFile>.suffix","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"w3igy8vmAmuAmk9cZVA9","level":3,"time":"2020-08-28T07:34:29+00:00","remoteAddr":"<IPAddr>","user":"<User>","app":"PHP","method":"PUT","url":"/<PathToNextcloud>/remote.php/webdav/<ServerAutoUploadDir>/<ChangedFile>.suffix","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"KfNVLauAnG0scZTulSI6","level":3,"time":"2020-08-28T07:34:29+00:00","remoteAddr":"<IPAddr>","user":"<User>","app":"PHP","method":"PROPFIND","url":"/<PathToNextcloud>/remote.php/webdav/<ServerAutoUploadDir>/<ChangedFile>.suffix","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"26RDkBsRdrwoBsDvNjl6","level":3,"time":"2020-08-28T07:34:29+00:00","remoteAddr":"<IPAddr>","user":"<User>","app":"PHP","method":"GET","url":"/<PathToNextcloud>/index.php/apps/files/api/v1/thumbnail/256/256/<ServerAutoUploadDir>/<ChangedFile>.suffix","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"NalS6wChD6Be9VnuG5SE","level":3,"time":"2020-08-28T07:34:38+00:00","remoteAddr":"<IPAddr>","user":"--","app":"PHP","method":"GET","url":"/<PathToNextcloud>/index.php/204","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"kKWYtV4pOy6XMF2EperJ","level":3,"time":"2020-08-28T07:34:38+00:00","remoteAddr":"<IPAddr>","user":"--","app":"PHP","method":"GET","url":"/<PathToNextcloud>/index.php/204","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"1npoRREcSc3YPAGVFbK5","level":3,"time":"2020-08-28T07:34:38+00:00","remoteAddr":"<IPAddr>","user":"--","app":"PHP","method":"GET","url":"/<PathToNextcloud>/index.php/204","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"3v5Gd7Z1FLVbq2G6WcPl","level":3,"time":"2020-08-28T07:34:38+00:00","remoteAddr":"<IPAddr>","user":"--","app":"PHP","method":"GET","url":"/<PathToNextcloud>/index.php/204","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"y7a5PCYId09HRaVSCZtG","level":3,"time":"2020-08-28T07:34:38+00:00","remoteAddr":"<IPAddr>","user":"--","app":"PHP","method":"GET","url":"/<PathToNextcloud>/index.php/204","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"fviNEnl6K6WDvvTyhnnU","level":3,"time":"2020-08-28T07:34:38+00:00","remoteAddr":"<IPAddr>","user":"--","app":"PHP","method":"GET","url":"/<PathToNextcloud>/index.php/204","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"VOpOQRtulWVOKavIZ3BW","level":3,"time":"2020-08-28T07:34:38+00:00","remoteAddr":"<IPAddr>","user":"--","app":"PHP","method":"GET","url":"/<PathToNextcloud>/index.php/204","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
{"reqId":"Of0vUkLxZJAxAHzdRQ8H","level":3,"time":"2020-08-28T07:34:38+00:00","remoteAddr":"<IPAddr>","user":"--","app":"PHP","method":"GET","url":"/<PathToNextcloud>/index.php/204","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at /<NextcloudInstallDir>/3rdparty/patchwork/utf8/src/Patchwork/Utf8/Bootup/intl.php#18","userAgent":"Mozilla/5.0 (Android) Nextcloud-android/3.13.0","version":"19.0.1.1"}
```

**Comment by tobiasKaminsky** [comment]
Duplicate of #5755
If not, please re-open.
